modules/oe_theme_content_consultation/oe_theme_content_consultation.module

```diff
@@ -91,3 +91,14 @@ function oe_theme_content_consultation_preprocess_field(&$variables, $hook): voi
     $variables['attributes']['class'][] = 'ecl-editor';
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ *
+ * Renders Publication teaser view mode inside Document reference entity.
+ */
+function oe_theme_content_consultation_preprocess_field__oe_document_reference__oe_publication(&$variables): void {
```

Comment:
Instead of rendering entities manually I would propose to create oe_theme_content_entity_document_reference and provide core.entity_view_display.* configurations for the Document and Publication bundles. So it can be reused in other CTs. See oe_theme_content_entity_contact.